ios/src/TangramMap/TGMapViewController.h

```diff
@@ -0,0 +1,731 @@
+
+ @note If no feature have been selected, `feature` will be `nil`, and `position` `(0, 0)`.
+ */
+- (void)mapView:(nonnull TGMapViewController *)mapView didSelectFeature:(nullable NSDictionary *)feature atScreenPosition:(CGPoint)position;
+ /**
+ Always called after the method `-[TGMapViewController pickLabelAt:]` is called on the map view.
+
+ @param mapView a pointer to the map view
+ @param labelPickResult the label pick result object containing information about the label being picked, may be `nil`
+ @param position the screen position at which the feature has been selected
+
+ @note If no feature have been selected, `feature` will be `nil`, and `position` `(0, 0)`.
+ */
+- (void)mapView:(nonnull TGMapViewController *)mapView didSelectLabel:(nullable TGLabelPickResult *)labelPickResult atScreenPosition:(CGPoint)position;
+/**
+ Always called after the method `-[TGMapViewController pickMarkerAt:]` is called on the map view.
+
+ @param mapView a pointer to the map view
+ @param markerPickResult the marker pick result object containing information about the marker being picked, may be `nil`
+ @param position the screen position at which the feature has been selected
+
+ @note If no feature have been selected, `feature` will be `nil`, and `position` `(0, 0)`.
+ */
+- (void)mapView:(nonnull TGMapViewController *)mapView didSelectMarker:(nullable TGMarkerPickResult *)markerPickResult atScreenPosition:(CGPoint)position;
+/**
+ Called whenever the view did complete loading and building of all of the visible tiles of the current view.
+
+ @param mapView a pointer to the map view
+ */
+- (void)mapViewDidCompleteLoading:(nonnull TGMapViewController *)mapView;
+@end
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ `TGMapViewController` is a flexible and customizable map view, allowing you to embed
+ it inside a <a href="https://developer.apple.com/reference/glkit/glkview">GLKView</a> component.
+ This view provides default gesture handlers for tap, double tap, long press, pan,
+ pinch, rotate and shove.
+
+ The public interface provides dynamic map marker placement, change of camera view
+ settings, and map description changes through scene updates.
+
+ This view uses scenes descibed by the
+  <a href="https://mapzen.com/documentation/tangram/">Tangram YAML scene syntax</a>
+ allowing you to fully customize your map description using your own data.
+ Some already made basemap styles can be found
+ <a href="https://mapzen.com/documentation/cartography/styles/">here</a> using Mapzen
+ data sources.
+
+ To use basemap styles you can <a href="https://mapzen.com/developers/sign_in">sign up for
+ an API key</a> and load it through your application:
+
+ ```swift
+ let sceneURL = "https://mapzen.com/carto/walkabout-style-more-labels/walkabout-style-more-labels.yaml";
+ view.loadSceneFile(sceneURL, sceneUpdates: [ TGSceneUpdate(path: "sources.mapzen.url_params", value: "{ api_key: \(YOUR_API_KEY) }") ]);
```

Comment:
I would update to only objective-c since I think that both are a little too much to give the user any additional information on the API usage, but let me know your preference, only swift, only objective-c or both?